src/web3/contracts.ts

```diff
@@ -109,6 +116,30 @@ async function waitForContractKit(tries: number) {
   return contractKit
 }
 
+export function* getViemWallet(chain: Chain) {
+  if (viemWallets.has(chain)) {
+    return viemWallets.get(chain) as ViemWallet
+  }
+  const walletAddress = yield* select(walletAddressSelector)
+  const mnemonic = yield* call(getStoredMnemonic, walletAddress)
```

Comment:
this would require a PIN right? does it open the pin screen automatically?

Reply:
yes it would, but refactoring to get the pin more directly